lib/offline_sort/merger.rb

```diff
@@ -0,0 +1,54 @@
+require 'forwardable'
+
+module OfflineSort
+  class Merger
```

Comment:
I'm not 100% sure, but I think if you (1) include `Enumerable` (https://ruby-doc.org/core-2.7.1/Enumerable.html) and (2) make `enumerator` a private method, then this `Merger` will quack kind of like an array and will let us call things like `chunk_while`, `to_a` etc on it. I think this is the contract that the gem originally had (because it was returning a real enumerator) and so may be easier to integrate into our codebase if we maintain that.